src/bundle.ts

```diff
@@ -104,68 +104,65 @@ export class Bundler extends Transform {
     return this.allFragments.indexOf(file.path) !== -1;
   }
 
-  _buildBundles(): Promise<Map<string, string>> {
-    return this._getBundles().then((bundles) => {
-      let sharedDepsBundle = (this.shell)
-          ? urlFromPath(this.root, this.shell)
-          : this.sharedBundleUrl;
-      let sharedDeps = bundles.get(sharedDepsBundle) || [];
-      let promises: Promise<any>[] = [];
-
-      if (this.shell) {
-        let shellFile = this.analyzer.getFile(this.shell);
-        console.assert(shellFile != null);
-        let newShellContent = this._addSharedImportsToShell(bundles);
-        shellFile.contents = new Buffer(newShellContent);
-      }
+  async _buildBundles(): Promise<Map<string, string>> {
+    let bundles = await this._getBundles();
```

Comment:
Here and elsewhere: preferred style is `const` by default, `let` only when necessary.